crossbuilder

```diff
@@ -958,8 +1140,21 @@ check_command_parameter_count () {
 
 detect_package () {
     if test -e debian/changelog ; then
-        PACKAGE=`dpkg-parsechangelog --show-field Source`
-        PACKAGE_VERSION=`dpkg-parsechangelog --show-field Version`
+        if which dpkg-parsechangelog > /dev/null 2>&1 ; then
+            PACKAGE=`dpkg-parsechangelog --show-field Source`
+            PACKAGE_VERSION=`dpkg-parsechangelog --show-field Version`
+        else
+              # Get package name and version parsing the debian/changelog file using grep.
+              local RES=($(grep -P '^[^ ]+ \([^)]+\) [^;]+; urgency=.*$' debian/changelog | head -n1 | sed 's/^\([^ ]\+\) (\([^)]\+\)).*/\1 \2/'))
+              PACKAGE=${RES[0]}
+              PACKAGE_VERSION=${RES[1]}
+              #echo "PACKAGE=\"${PACKAGE}\""
+              #echo "PACKAGE_VERSION=\"${PACKAGE_VERSION}\""
+              if [ -z $PACKAGE ] || [ -z $PACKAGE_VERSION ] ; then
```

Comment:
you could actually move it one level up, just in case the changelog is bogus and `dpkg-changelog` fails too.

Reply:
done